src/Compilers/CSharp/Test/Semantic/Semantics/ForEachTests.cs

```diff
@@ -2865,6 +2865,137 @@ static void Main(string[] args)
             );
         }
 
+        [WorkItem(11387, "https://github.com/dotnet/roslyn/issues/11387")]
+        [Fact]
+        public void StringNotIEnumerable()
+        {
+            var source1 =
+@"namespace System
+{
+    public class Object { }
+    public struct Void { }
+    public class ValueType { }
+    public struct Boolean { }
+    public struct Int32 { }
+    public struct Char { }
+
+    public class String 
+    {
+        public int Length => 2;
+
+        [System.Runtime.CompilerServices.IndexerName(""Chars"")]
+        public char this[int i] => 'a';
```

Comment:
Yes. IndexerName  is needed for the indexer to be emitted as Special Member "Chars_get". 
To keep indexer an indexer (and not just a parameterized property), compiler also requires DefaultMemberAttribute as well.